packages/web3-eth/src/utils/transaction_builder.ts

```diff
@@ -238,7 +236,21 @@ export async function defaultTransactionBuilder<ReturnType = Transaction>(option
 				ETH_DATA_FORMAT,
 			)),
 		};
-
+	if (
+		(isNullish(populatedTransaction.gas) || isNullish(populatedTransaction.gasLimit)) &&
```

Comment:
```suggestion
		(isNullish(populatedTransaction.gas) && isNullish(populatedTransaction.gasLimit)) &&
```
if both gas and gasLimit are not available then we need to set gas, else one of this should be used to set gas.